src/execution/__tests__/executor-test.js

```diff
@@ -306,11 +307,65 @@ describe('Execute: Handles basic execution tasks', () => {
     const field = operation.selectionSet.selections[0];
     expect(resolvedInfo).to.deep.include({
       fieldNodes: [field],
-      path: { prev: undefined, key: 'result' },
+      path: { prev: undefined, key: 'result', parentType: 'Test' },
       variableValues: { var: 'abc' },
     });
   });
 
+  it('populates path correctly with complex types', () => {
+    let path;
+    const a = new GraphQLObjectType({
+      name: 'A',
+      fields: {
+        test: {
+          type: GraphQLString,
+          resolve(_val, _args, _ctx, info) {
+            path = info.path;
+          },
+        },
+      },
+    });
+    const testUnion = new GraphQLUnionType({
+      name: 'TestUnion',
```

Comment:
```suggestion
      name: 'SomeUnion',
```
We slowly migrating all the tests to use same naming convention for test types: `SomeObject`, `AnotherObject`, `SomeUnion`, `SomeScalar`, etc.

Reply:
and the same goes for variable names `someUnion`.